pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -89,6 +89,26 @@ func NodePoolEventHandler(c client.Client) handler.EventHandler {
 	})
 }
 
+// NodeClassEventHandler is a watcher on v1beta1.NodeClaim that maps NodeClass to NodeClaims based
+// on the nodeClassRef and enqueues reconcile.Requests for the NodeClaim
+func NodeClassEventHandler(c client.Client) handler.EventHandler {
```

Comment:
How is this going to work when someone doesn't set the apiVersion or the kind today? We need some handling to ensure that we work with the default apiVersion/kind appropriately

Reply:
Today, you will get an empty string back from some of the NodeClaims, and then you aren't going to requeue appropriately on it.

Reply:
I was thinking that we could tell user to set the apiVersion and Kind to get better response for drift detection. Since it's not a bug to have an eventually consistent drift detection on NodeClass, I was thinking this would work really nicely once we default and require all fields in the NodeClassRef